autoextract_poet/page_inputs.py

```diff
@@ -0,0 +1,59 @@
+from typing import ClassVar, List, Optional, Type
+
+import attr
+
+from autoextract_poet.items import (
+    Article,
+    Item,
+    Product,
+)
+
+
+@attr.s(auto_attribs=True)
+class ResponseData:
+    """Container for AutoExtract response data.
+
+    Should not be used directly by providers.
+    Use derived classes like ArticleResponseData and similar.
+
+    API responses are wrapped in a JSON array
+    (this is to facilitate query batching).
+
+    https://doc.scrapinghub.com/autoextract.html#responses
+    """
+
+    item_class: ClassVar[Type[Item]]
+    item_key: ClassVar[str]
+
+    data: dict
+
+    def to_items(self) -> Optional[List[Item]]:
+        return self.item_class.from_list(
+            [
+                query_result[self.item_key]
+                for query_result in self.data
+                if "error" not in query_result
+            ]
+        )
+
+
+@attr.s(auto_attribs=True)
+class ArticleResponseData(ResponseData):
```

Comment:
A proposal, for naming- up to discussion. What do you think about calling it AutoextractArticleData, or just ArticleData? 

* ResponseData means a very different thing in web_poet - url+html, so having ResponseData in the name can be confusing.
* I'm on fence about having Autoextract in the name.

Reply:
"Data" would convey that it is still raw data returned by the API, not an item.

Reply:
TIL "I'm on fence". Agree with removing `Response` from the name. Regarding Autoextract, I'm frozen over the fence. 
Ideas. If we decide not to include it in the name, I think we should then have a module named `autoextract`, so that you can have do the following:

```py
from autoextract_poet import autoextract
from manualextract_poet import manualextract

def my_function(auto_product: autoextract.ProductData, manual_product: manualextract.ProductData:
  ...
```

Reply:
I went with the `AutoExtractArticleData` variant because it's not a really long name (and based on what we currently have on our documentation page, names shouldn't get too much larger) and it could avoid some confusion when importing page inputs from different places.

Of course, users could make something like:

```python
from autoextract_poet import page_inputs as autoextract_page_inputs
autoextract_page_inputs.ArticleData
```

or even

```python
from autoextract_poet.page_inputs import (
    ArticleData as AutoExtractArticleData,
    ProductData as AutoExtractProductData,
)
```

But I think we could give them a hand and avoid this ambiguity since now :)